python/cudf/cudf/core/column/string.py

```diff
@@ -485,10 +485,11 @@ def join(
             If array-like, the string at a position is used as a
             delimiter for corresponding row of the list entries.
         string_na_rep : str, default None
-            This character will take the place of any null strings
-            (not empty strings) in the Series.
-            If ``string_na_rep`` is ``None``, it defaults to empty
-            space "".
+            This character will take the place of null strings
+            (not empty strings) in the Series but will be considered
+            only if the Series contains at least one non-null string.
+            If ``string_na_rep`` is ``None`` or the Series contains only
+            ``None``, it defaults to empty space "".
```

Comment:
It would be really helpful to document this behaviour in an example.

Reply:
Thanks @shwina. There are already several examples demonstrating such behavior---null replacement if there are both Null and valid strings, and empty return string (ignore null replacement) if there are only Null.

Reply:
I meant in the [`Examples`](https://github.com/rapidsai/cudf/blob/07b8c9c646f2f15acb0821b42feda462880bd745/python/cudf/cudf/core/column/string.py#L514) section of the docstring for this function. I don't think we have an example for the "all nulls" case described.

Reply:
Sure. Done.